code-studio/src/js/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -0,0 +1,567 @@
+  },
+
+  propTypes: {
+    workshop: React.PropTypes.shape({
+      id: React.PropTypes.number.isRequired,
+      facilitators: React.PropTypes.array.isRequired,
+      location_name: React.PropTypes.string.isRequired,
+      location_address: React.PropTypes.string.isRequired,
+      capacity: React.PropTypes.number.isRequired,
+      workshop_type: React.PropTypes.string.isRequired,
+      course: React.PropTypes.string.isRequired,
+      subject: React.PropTypes.string,
+      notes: React.PropTypes.string,
+      sessions: React.PropTypes.array.isRequired
+    }),
+    onSaved: React.PropTypes.func,
+    readOnly: React.PropTypes.bool
+  },
+
+  getInitialState: function () {
+    let initialState = {
+      errors: [],
+      shouldValidate: false,
+      facilitators: [],
+      location_name: '',
+      location_address: '',
+      capacity: '',
+      workshop_type: '',
+      course: '',
+      subject: '',
+      notes:'',
+      sessions: [{placeholderId: '_0'}],
+      destroyedSessions: [],
+      availableFacilitators: [],
+      showSaveConfirmation: false
+    };
+
+    if (this.props.workshop) {
+      initialState = _.merge(initialState,
+        _.pick(this.props.workshop, [
+        'facilitators',
+        'location_name',
+        'location_address',
+        'capacity',
+        'workshop_type',
+        'course',
+        'subject',
+        'notes'
+        ])
+      );
+      initialState.sessions = this.prepareSessionsForForm(this.props.workshop.sessions);
+      this.loadAvailableFacilitators(this.props.workshop.course);
+    }
+    return initialState;
+  },
+
+  componentDidMount: function () {
+    this.enableAutocompleteLocation();
+  },
+
+  componentWillUnmount: function () {
+    if (this.isGoogleMapsLoaded()) {
+      google.maps.event.clearInstanceListeners(this.autocomplete);
+    }
+    if (this.saveRequest) {
+      this.saveRequest.abort();
+    }
+    if (this.loadWorkshopRequest) {
+      this.loadWorkshopRequest.abort();
+    }
+  },
+
+  componentWillReceiveProps: function (nextProps) {
+    if (nextProps.readOnly && !this.props.readOnly) {
+      this.setState(this.getInitialState());
+    }
+  },
+
+  componentDidUpdate: function () {
+    this.enableAutocompleteLocation();
+  },
+
+  loadAvailableFacilitators: function (course) {
+    this.loadWorkshopRequest = $.ajax({
+      method: "GET",
+      url: `/api/v1/pd/course_facilitators?course=${course}`,
+      dataType: "json"
+    }).done(function (data) {
+      this.setState(_.set(this.state, 'availableFacilitators', data));
+    }.bind(this));
+  },
+
+  isGoogleMapsLoaded: function () {
+    return (typeof google === 'object' && typeof google.maps === 'object');
+  },
+
+  enableAutocompleteLocation: function () {
+    if (!this.autocomplete && this.isGoogleMapsLoaded()) {
+      this.autocomplete = new google.maps.places.Autocomplete($(ReactDOM.findDOMNode(this)).find('.location-autocomplete')[0]);
```

Comment:
In your jsx, you can add `ref='locationAutocomplete'` at which point this can just be `this.refs.locationAutocomplete`. Not something that needs to block checkin.